app/screens/terms_of_service/terms_of_service.js

```diff
@@ -0,0 +1,247 @@
+        id: 'accept-terms-of-service',
+        showAsAction: 'always',
+    };
+
+    constructor(props, context) {
+        super(props);
+
+        this.state = {
+            error: null,
+            loading: true,
+            serverError: null,
+            termsId: '',
+            termsText: '',
+        };
+
+        this.rightButton.title = context.intl.formatMessage({id: 'terms_of_service.agreeButton', defaultMessage: 'I Agree'});
+        this.leftButton.icon = props.closeButton;
+
+        props.navigator.setOnNavigatorEvent(this.onNavigatorEvent);
+        this.setNavigatorButtons();
+    }
+
+    componentDidMount() {
+        this.getTerms();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.theme !== prevProps.theme) {
+            setNavigatorStyles(this.props.navigator, this.props.theme);
+        }
+    }
+
+    setNavigatorButtons = (enabled = true) => {
+        const buttons = {
+            leftButtons: [{...this.leftButton, disabled: !enabled}],
+            rightButtons: [{...this.rightButton, disabled: !enabled}],
+        };
+
+        this.props.navigator.setButtons(buttons);
+    };
+
+    getTerms = () => {
+        const {intl} = this.context;
+
+        this.setState({
+            termsId: '',
+            termsText: '',
+            loading: true,
+            serverError: null,
+        });
+
+        getTermsOfService(
+            (data) => {
+                this.setState({
+                    termsId: data.id,
+                    termsText: data.text,
+                    loading: false,
+                });
+            },
+            () => {
+                this.setState({
+                    loading: false,
+                    serverError: intl.formatMessage({
+                        id: 'terms_of_service.api_error',
+                        defaultMessage: 'Unable to complete the request. If this issue persists, contact your System Administrator.',
+                    }),
+                });
+
+                // TODO: Show an error message on login screen
+            }
+        );
+    };
+
+    handleAcceptTerms = () => {
+        this.setNavigatorButtons(false);
+        this.registerUserAction(
+            true,
+            () => {
+                this.setNavigatorButtons(true);
+                this.props.navigator.dismissModal({
+                    animationType: 'slide-down',
+                });
+            }
+        );
+    };
+
+    handleRejectTerms = () => {
+        const {logout} = this.props.actions;
+        this.setNavigatorButtons(false);
+        this.registerUserAction(
+            false,
+            () => {
+                this.setNavigatorButtons(true);
+                this.props.navigator.dismissAllModals();
+                InteractionManager.runAfterInteractions(logout);
+
+                // TODO: Show an error message on login screen
+            }
+        );
+    };
+
+    registerUserAction = (accepted, success) => {
```

Comment:
you don't need this function and more in here is that you set the loading state as **true** so that means that while you are sending the acceptance or rejection of the terms the text in the screen disappears and gets replaced with a loading indicator, is that expected?